core/examples/http_bench.js

```diff
@@ -1,11 +1,58 @@
 // This is not a real HTTP server. We read blindly one time into 'requestBuf',
 // then write this fixed 'responseBuf'. The point of this benchmark is to
 // exercise the event loop in a simple yet semi-realistic way.
-const OP_LISTEN = 1;
-const OP_ACCEPT = 2;
-const OP_READ = 3;
-const OP_WRITE = 4;
-const OP_CLOSE = 5;
+class Op {
```

Comment:
This class needs a bit more work. What I had in mind is that we create `JsonOp` which encapsulates json serialization/deserialization for ops. Then you create your own `JsonOp`s and during startup instead of calling `Deno.core.recv(...)` we'll call `Deno.core.initOps()` which will synchronize all ops between JS and Rust.

Reply:
Side note: Given that this is pretty elegant API IMHO that allows for "automagical" registration of ops, I'm still worried about performance of this solution because each op is effectively a closure. But let's see in benchmarks first.

Reply:
So I thought I nailed it, but this automagical registration might be harder to pull off than I expected. I got it working fine for `deno_core_http_bench`, but I can't get it working with CLI. The problem is snapshotting - ops are registered during snapshot build, but upon running new deno process shared queue is loaded during isolate startup and I guess the one from snapshot is ignored. 

For more detail see #3004

Reply:
I got another idea how to implement that, working on it